### PR TITLE
Enhance query filters to handle Uuid and enable analysis graph to use filters.

### PR DESCRIPTION
**Pull request by JimFuller-RedHat**
Enhances query.rs to be able to perform comparisons with **Uuid** when supplied via inline query params.

Enabling the analysis graph (api/v1/analysis) to filter on **sbom_id**, **node_id** and exact **name**.
```
api/v1/analysis/root-component?q=name=A&node_id=SPDXRef-A&sbom_id=urn:uuid:019289ff-232c-7780-b0d5-22b6e5dc4be5
```
where we retain **contains** style search with:
```
api/v1/analysis/root-component?q=libcurl
```
to filter exact match use:
```
api/v1/analysis/root-component?q=name=libcurl
```

and similar for dep endpoints eg.
```
api/v1/analysis/dep?q=name=A&node_id=SPDXRef-A&sbom_id=urn:uuid:019289ff-232c-7780-b0d5-22b6e5dc4be5
api/v1/analysis/dep?q=libcurl
api/v1/analysis/dep?q=name=libcurl
```

**Note** - In real usage - the above inline _query name=values&name=value_ should always be escaped.

Most of the 'moving parts' of this feature is due to querying both an ORM and non ORM entity at the same time (sbom_node and analysis graph) ... kept most of the code changes within analysis graph.

### CODE REVIEW

**Review by ctron — COMMENTED** [review]
LGTM, but I'll defer to @jcrossley3

File: common/src/db/query.rs
Comment (jcrossley3) [inline common/src/db/query.rs:477]: Ideally, we'd have a test for this in this file. Can you change the type for `id` from `i32` to `Uuid` in the dummy advisory model at the bottom of this file, and re-run the tests? I expect you'll need to add some quotes around some of the expected where clauses.

If that's too big of a hassle, I'm happy to do it after this PR merges.

File: modules/analysis/src/service.rs
Comment (jcrossley3) [inline modules/analysis/src/service.rs:292]: Now I understand why you were asking how to get the parsed value out of `Query`. If we find ourselves wanting this in other places, we can refactor to hang this fn off `Query`.

Minor style nit: I don't love the "mutable hashmap plus for loop" pattern here. Smells like old java. :stuck_out_tongue_winking_eye:  Next time, consider calling `reduce` or `fold` on the `split`.

**Review by jcrossley3 — COMMENTED** [review]
Glad you figured it out!

File: common/src/db/query.rs
Comment (JimFuller-RedHat) [reply common/src/db/query.rs:477]: will add!

File: modules/analysis/src/service.rs
Comment (JimFuller-RedHat) [reply modules/analysis/src/service.rs:292]: ah ya, I made this loop explicit as I could not figure out why things were not working -  then grokked the need to strip urn:uuid: for the graph query bit ... will rewrite, thx!!

File: common/src/db/query.rs
Comment (jcrossley3) [inline common/src/db/query.rs:822]: Thanks!

File: modules/analysis/src/service.rs
Comment (jcrossley3) [inline modules/analysis/src/service.rs:292]: I'm not sure this is more clear. :smile:

File: modules/analysis/src/service.rs
Comment (jcrossley3) [inline modules/analysis/src/service.rs:409]: This lump of code is repeated below, which is crying out for a refactor into a single, common fn. If it were me, I'd probably rename/modify `convert_query_to_hashmap` to take both a `Query` and a `Node` and then return a `bool`, and call it here. 

I'd prolly use regex's instead of the temporary hashmap, too, so feel free to ignore me. :smile:

**Review by jcrossley3 — COMMENTED** [review]
Did you want to fix the `graph_read_gaurd` typo while you're in there?

File: modules/analysis/src/service.rs
Comment (JimFuller-RedHat) [reply modules/analysis/src/service.rs:409]: this area may evolve depending on @dejanb usage ... so will leave it for now.

**Review by jcrossley3 — APPROVED** [review]

**Pull request merged**
